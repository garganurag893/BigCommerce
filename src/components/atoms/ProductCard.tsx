import { FC } from 'react'
import { ProductDetails } from '../../types'

interface ProductCardProps {
  product: ProductDetails
  isSelected?: boolean
  onClick?: () => void
}

const ProductCard: FC<ProductCardProps> = ({ product, isSelected = false, onClick }) => {
  const { image, title, rating, price } = product || {}
  return (
    <li
      data-testid='product-card'
      onClick={onClick}
      className={`mx-3 my-6 flex flex-col justify-between max-w-xxs sm:w-72 rounded-lg overflow-hidden cursor-pointer hover:shadow-lg hover:scale-105 duration-300 ${
        isSelected ? 'bg-black shadow-xl' : ''
      }`}
    >
      <div>
        <div className='relative h-64 w-full'>
          <img src={image} alt={title} className='object-contain h-full w-full bg-white' />
          <div className='absolute bottom-2 right-2 bg-gray-300/40 flex flex-row items-center px-1 rounded-sm'>
            <img src='/assets/png/star.png' alt='Star' height={16} width={16} />
            <p className='ml-2 font-semibold text-sm'>{rating.rate}</p>
          </div>
        </div>
        <p className={`font-bold text-base p-2 ${isSelected ? 'text-white' : 'text-black'}`}>
          {title}
        </p>
      </div>
      <p className={`p-2 text-xs font-bold ${isSelected ? 'text-white' : 'text-black'}`}>
        $ {price}
      </p>
    </li>
  )
}

export default ProductCard
