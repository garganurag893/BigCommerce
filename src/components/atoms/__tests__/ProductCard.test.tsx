import { render, fireEvent } from '@testing-library/react'

import ProductCard from '../ProductCard'

const productData = {
    id: 13,
    title: 'Mens Cotton Jacket',
    price: 55.99,
    description: 'Great outerwear jackets for Spring',
    category: 'mens clothing',
    image: 'https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg',
    rating: { rate: 4.7, count: 500 },
  }

describe('Test ProductCard Component', () => {
  it('Check for ProductCard tag in DOM', () => {
    const onClickHandler = jest.fn()
    const { getByTestId } = render(
      <ProductCard
        product={productData}
        onClick={onClickHandler}
      />,
    )

    expect(getByTestId('product-card')).toBeInTheDocument()
    fireEvent.click(getByTestId('product-card'))

    expect(onClickHandler).toHaveBeenCalledTimes(1)
  })

  it('ProductCard Snapshot Test', () => {
    const onClickHandler = jest.fn()
    const { container } = render(
      <ProductCard
        product={productData}
        onClick={onClickHandler}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it('ProductCard Selectec Snapshot Test', () => {
    const onClickHandler = jest.fn()
    const { container } = render(
      <ProductCard
        product={productData}
        isSelected
        onClick={onClickHandler}
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
