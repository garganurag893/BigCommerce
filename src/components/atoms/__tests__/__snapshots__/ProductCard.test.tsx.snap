// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProductCard Component ProductCard Selectec Snapshot Test 1`] = `
<div>
  <li
    class="mx-3 my-6 flex flex-col justify-between max-w-xxs sm:w-72 rounded-lg overflow-hidden cursor-pointer hover:shadow-lg hover:scale-105 duration-300 bg-black shadow-xl"
    data-testid="product-card"
  >
    <div>
      <div
        class="relative h-64 w-full"
      >
        <img
          alt="Mens Cotton Jacket"
          class="object-contain h-full w-full bg-white"
          src="https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg"
        />
        <div
          class="absolute bottom-2 right-2 bg-gray-300/40 flex flex-row items-center px-1 rounded-sm"
        >
          <img
            alt="Star"
            height="16"
            src="/assets/png/star.png"
            width="16"
          />
          <p
            class="ml-2 font-semibold text-sm"
          >
            4.7
          </p>
        </div>
      </div>
      <p
        class="font-bold text-base p-2 text-white"
      >
        Mens Cotton Jacket
      </p>
    </div>
    <p
      class="p-2 text-xs font-bold text-white"
    >
      $ 
      55.99
    </p>
  </li>
</div>
`;

exports[`Test ProductCard Component ProductCard Snapshot Test 1`] = `
<div>
  <li
    class="mx-3 my-6 flex flex-col justify-between max-w-xxs sm:w-72 rounded-lg overflow-hidden cursor-pointer hover:shadow-lg hover:scale-105 duration-300 "
    data-testid="product-card"
  >
    <div>
      <div
        class="relative h-64 w-full"
      >
        <img
          alt="Mens Cotton Jacket"
          class="object-contain h-full w-full bg-white"
          src="https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg"
        />
        <div
          class="absolute bottom-2 right-2 bg-gray-300/40 flex flex-row items-center px-1 rounded-sm"
        >
          <img
            alt="Star"
            height="16"
            src="/assets/png/star.png"
            width="16"
          />
          <p
            class="ml-2 font-semibold text-sm"
          >
            4.7
          </p>
        </div>
      </div>
      <p
        class="font-bold text-base p-2 text-black"
      >
        Mens Cotton Jacket
      </p>
    </div>
    <p
      class="p-2 text-xs font-bold text-black"
    >
      $ 
      55.99
    </p>
  </li>
</div>
`;
